### PR TITLE
Readme: more detailed build instructions

### PR DESCRIPTION
**Pull request by folletto**
I've setup the Android project from scratch, and while doing so I also noted down all the steps. With that data, I then updated the readme file so it's easier to get started with the project. :)

For reference this is written having in mind:

* Someone with basic dev/Android knowledge: they should be able to set this up without extra help, but they aren't experts at all.
* Experts: they should be able to read this quickly, without getting frustrated by a description that is TOO detailed (and thus, boring) for them.

 

_(screenshot out of date, kept for reference)_

screen shot 2017-10-09 at 18 44 17

Notes:

1. I'd like if possible to add an extra script that could replace some of the manual terminal steps above. I guess the best scenario would be to add a new file under `/tools`. Is this assumption correct?
2. I haven't yet done the OAuth 2 series of steps, I'll probably do another update (either here or in a new PR if this gets merged sooner) for that.

### CODE REVIEW

**Comment by mzorz** [comment]
Nice @folletto ! thank you for making this effort 🙇 

> I'd like if possible to add an extra script that could replace some of the manual terminal steps above. I guess the best scenario would be to add a new file under /tools. Is this assumption correct?

Yes that'd be right, but I doubt having a script that does a manual copy of a file add too much value, also it's good to go through it manually so a newcomer can understand what the project is parameterized like (as opposed as doing that automagically)

I like having the updated step-by-step guide to set this up though! 🎉

**Comment by folletto** [comment]
> Yes that'd be right, but I doubt having a script that does a manual copy of a file add too much value, also it's good to go through it manually so a newcomer can understand what the project is parameterized like (as opposed as doing that automagically)

True, I'm just trying to consider different perspective: if someone is installing just to test, might not be very interested in that.

The idea for the script would be:

1. Check for JDK and Android Studio being set up (and maybe even trigger download & install from command line if there's a way to do that easily)
2. Create `local.properties` automatically by checking if Android SDK is installed in a known path (I haven't found a way to retrieve this programmatically)
3. Copy `gradle.properties` if not existing already
4. Maybe setup already a device, but I kinda failed so far from command line, so maybe it's not possible.
5. Maybe even setup OAuth 2 automatically, but again seems not possible.

It's not much, but I guess it's also one of the things that once there stuff starts getting added... :D

**Comment by aforcier** [comment]
My two cents: The instructions for setting up Android Studio and the JDK feel a bit out of place.

Could we do something like [Calypso does](https://github.com/Automattic/wp-calypso#getting-started) instead, and add a link to a separate 'detailed instructions' doc that walks you through those parts ("Need more detailed installation instructions? We have them.")?

**Comment by folletto** [comment]
> Could we do something like Calypso does instead [..]

Which states:

> (1) Make sure you have git, node, and npm installed.

;) 

If you prefer we can collapse them on one line.

May I ask however why you feel it's out of place to specify all the requirements to run something?

**Comment by mzorz** [comment]
Regarding the script only:

I still think it’s not going to add a lot of value, and if you follow the steps as described you don’t need to get very much involved into the details; i.e it will just work, and it makes you aware that a `local.properties`  file exists and needs be modified, and a `gradle.properties` exist and needs be there as well. As opposed to having something that obscures the process (even for a tiny bit) more.

Let's put it like this: if you're newbie, copying a file is simple a step as it can be but still doesn't hide anything. If you're an expert, then you will want to know which parameters are important for the project.

**Comment by folletto** [comment]
Ok, let's skip the installer (for now). 👍

**Comment by aforcier** [comment]
> (1) Make sure you have git, node, and npm installed.

I think 'make sure so-and-so' is installed is fair enough, but adding many steps that are requirements for running any Android project is what feels unnecessary for the top-level `README`. Calypso doesn't tell you _how_ to install those things, is what I was getting at.

> May I ask however why you feel it's out of place to specify all the requirements to run something?

It makes the `README` longer, and obscures the only parts that are actually necessary if you already have your general Android dev environment set up. Again, this is my own reaction, and I tend to prefer shorter `README`s that make the important information and let you dive further if needed.

**Comment by folletto** [comment]
Ok, understood. Yes, I understand the worry of a long readme quite well, and I thought I got a good balance here as the update is just a few lines longer and gains lots of clarity from the checklist approach — clearly not. :)

I'll rework the steps.

**Comment by folletto** [comment]
Ok, updated — 4 less bullet points and shorter wording for the more obvious parts. Now it's almost identical in structure with Calypso — it has longer bullet points because Calypso does most of the work inside the build script on first run, but as we don't want to do it we need a couple extra bullets. 

I also tweaked the language to acknowledge it might be read by an expert with everything already set up.

screen shot 2017-10-09 at 19 32 24

**Comment by maxme** [comment]
I think you don't need step 3 (create the `local.properties`) if you use Android Studio (AS will generate the file when you import the project for the first time). 

When we wrote this, we specified it that way (cli first) because we didn't want to force people using AS, but today things are a bit different and AS is a must have.

**Comment by folletto** [comment]
Confirmed, `local.properties` was auto-generated by Android Studio! Nice! 

screen shot 2017-10-10 at 12 51 02

**Comment by folletto** [comment]
Ok, updated:

screen shot 2017-10-10 at 13 07 57

**Comment by malinajirka** [comment]
This looks great:)! When you get to the OAuth part, make sure to note, that the app requires a white listed oauth ID and Secret in order to authenticate to WordPress.com.

**Comment by aforcier** [comment]
> When you get to the OAuth part, make sure to note, that the app requires a white listed oauth ID and Secret in order to authenticate to WordPress.com.

Hmmm, unless something has changed (broken), this shouldn't be the case - anyone can generate an app ID and secret and use it, the difference (compared to a whitelisted set) is that they can only use it to login to the WordPress.com account the ID/secret belong to.

Is that not working?

**Comment by malinajirka** [comment]
> Hmmm, unless something has changed (broken), this shouldn't be the case - anyone can generate an app ID and secret and use it, the difference (compared to a whitelisted set) is that they can only use it to login to the WordPress.com account the ID/secret belong to.
Is that not working?

Hmm, I talked with @aerych on slack about it a few days ago. As far as I understood the white listed oauth ID is required. We both tried to create a new app ID and login but an error always occurred - "Client cannot use 'password' grant_type".

**Comment by aforcier** [comment]
> Hmm, I talked with @aerych on slack about it a few days ago. As far as I understood the white listed oauth ID is required. We both tried to create a new app ID and login but an error always occurred - "Client cannot use 'password' grant_type".

This is interesting - I'll look into this and follow up 👍.

**Comment by aerych** [comment]
Chiming in.  :)  
When we tested several days ago it was the case that self-generated client ID and secrets were failing to authenticate a user.  @aforcier and I have both performed another round of tests and we've both successfully been able to use self-generated oauth credentials.   Its not clear why this was failing previously. :(  Alex raises the possibility that something might have changed/been broken server-side that has since been resolved. 
Suggesting we skip mentioning a need to white-list credentials until we sort out what's going on.

**Comment by folletto** [comment]
The OAuth part will be done separately — so if it's ok, I'll merge this first. :)

**Comment by hypest** [comment]
I see that the reviews have concluded and this seems ready to merge. 🚢 it!
